### DISCUSSION TITLE
how to call python code from javascript?

### DISCUSSION THREAD

**Discussion by beauxq** [pyscript/pyscript#2268] [Q&A]
The user guide in the documentation says
> The FFI is bi-directional ~ it also enables JavaScript to access the power of Python.

But I can't find any documentation or examples on how to call python code from javascript.

**Comment by WebReflection** [comment]
Let's say you have this python code:

```python
import binascii
import js

js.binascii = binascii
```

In JS then you can:

```js
console.log(binascii.crc32("a"));
```

You can also attach Python functionalities to nodes via events and whatnot, so that's the meaning of bidirectional: you have JS access from Python, you can expose Python to JS.

**edit** a live example: https://pyscript.com/@agiammarchi/binascii/latest?files=index.html
